JitsiMediaDevices.js

```diff
@@ -136,6 +136,14 @@ class JitsiMediaDevices {
             // Check using the Permissions API.
             this._permissionsApiSupported.then(supported => {
                 if (!supported) {
+                    // Workaround on Safari for audio input device
+                    // selection to work. Safari doesn't support the
+                    // permissions query.
+                    if (browser.isSafari()) {
```

Comment:
This would return true for any type of device. Doesn't it have bad side-effects?

Reply:
I have tested with all the audio/video devices I have and doesn't seem to have bad side-effects. My understanding is that we don't need to run permissions query for VIDEO_PERMISSION_NAME and AUDIO_PERMISSION_NAME anymore on any browsers, its legacy code before gUM wasn't standardized before and there were different implementations. Maybe check if its actually needed when we re-write the device selection logic. In the past I have done gUM calls directly w/o checking for permissions and haven't had issues.

Reply:
I have to add I might be wrong as well :)

Reply:
I think the original intent was to try and know beforehand if the permissions prompt was going to be displayed before asking.

Yeah, let’s keep it as is and reevaluate!